src/status_im2/contexts/wallet/select_address_to_watch/view.cljs

```diff
@@ -1,27 +1,59 @@
 (ns status-im2.contexts.wallet.select-address-to-watch.view
   (:require
-    [clojure.string :as string]
+    clojure.string
     [quo.core :as quo]
-    [quo.theme :as quo.theme]
     [re-frame.core :as re-frame]
-    [react-native.clipboard :as clipboard]
     [react-native.core :as rn]
     [react-native.safe-area :as safe-area]
     [reagent.core :as reagent]
+    [status-im2.constants :as constants]
     [status-im2.contexts.wallet.select-address-to-watch.style :as style]
+    [utils.debounce :as debounce]
     [utils.i18n :as i18n]
     [utils.re-frame :as rf]))
 
-(defn view-internal
+(defn- address-input
+  [input-value input-focused?]
+  (fn []
+    (let [valid-ens-or-address? (boolean (rf/sub [:wallet/valid-ens-or-address?]))
+          scanned-address       (rf/sub [:wallet/scanned-address])
+          on-change-text        (fn [text]
+                                  (let [starts-like-eth-address (re-matches
+                                                                 constants/regx-address-fragment
+                                                                 text)]
+                                    (when (and scanned-address (not= scanned-address text))
+                                      (rf/dispatch [:wallet/clean-scanned-address]))
+                                    (if starts-like-eth-address
+                                      (rf/dispatch [:wallet/fetch-address-suggestions text])
+                                      (rf/dispatch [:wallet/clean-local-suggestions]))
+                                    (reset! input-value text)))]
+      [quo/address-input
+       {:on-focus              #(reset! input-focused? true)
+        :on-blur               #(reset! input-focused? false)
+        :on-scan               #(rf/dispatch [:open-modal :scan-address])
+        :ens-regex             constants/regx-ens
+        :address-regex         constants/regx-address
+        :scanned-value         scanned-address
+        :on-detect-ens         #(debounce/debounce-and-dispatch
+                                 [:wallet/validate-ens %]
+                                 300)
+        :on-detect-address     #(debounce/debounce-and-dispatch
+                                 [:wallet/validate-address %]
+                                 300)
+        :on-change-text        on-change-text
+        :on-clear              #(rf/dispatch [:wallet/clean-scanned-address])
+        :valid-ens-or-address? valid-ens-or-address?}])))
+
+(defn f-view
   []
-  (let [top                 (safe-area/get-top)
-        bottom              (safe-area/get-bottom)
+  (let [bottom              (safe-area/get-bottom)
+        customization-color (rf/sub [:profile/customization-color])
         input-value         (reagent/atom "")
-        customization-color (rf/sub [:profile/customization-color])]
+        input-focused?      (reagent/atom false)]
```

Comment:
I noticed that the `input-focused?` value is not used inside the `address-input`. Are we planning to use it in future?

Reply:
whoops, I was being a bit blind there 😓 I'm going to remove this, and actually @briansztamfater I think on the send address screens we eventually should be using this as it will (should) handle that behaviour 
https://github.com/status-im/status-mobile/pull/17737

Reply:
will use the same approach on this page eventually too 👍